spec/services/pdf_parser_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'spec_helper'
+require_relative '../../app/services/pdf_parser'
+require './app/models/user'
+
+include AuthenticationSpecHelpers
+
+RSpec.describe PdfParser do
+  context 'when the PDF file is valid' do
+    let(:pdf_file) { Rack::Test::UploadedFile.new('spec/fixtures/sample.pdf', 'application/pdf') }
+    let!(:user) { authenticated }
+
+    it 'processes the uploaded PDF file' do
+      resident_count = Resident.count
+      PdfParser.new({tempfile: pdf_file}, user).load_residents
+
+      expect(Resident.count).to eq(resident_count + 23)
```

Comment:
You could also use something like `expect { do_the_thing }.to change(Resident, :count).by(23)` here, which I think is slightly more expressive.

Reply:
I think this comment still stands. You can do basically what you're doing here with just my suggestion. the `change` matcher checks the count before you start and then at the end.